### PR TITLE
Add a filter on recommendation

### PR DESCRIPTION
**Pull request by benjlevesque**
As suggested in #321, this PR adds a filter to hide containers that matches the VPA recommendation.

The proposed filters are:
- all: current behaviour
- any: show containers that don't meet recommendations
- guaranteed: show containers that don't meet guaranteed recommendations
- burstable: show containers that don't meet burstable recommendations

UI:

image

As I am a new user of goldilocks, let me know if those filters are not the correct ones.

### CODE REVIEW

**Comment by sudermanjr** [comment]
First glance this looks really great. Thanks for the contribution! 

I'll try to give it a more thorough review this week.

**Review by sudermanjr — COMMENTED** [review]
I think there might be an issue with the filter logic. Just testing functionally, I get this

```
▶ ./goldilocks summary --filter any | jq .
{
  "Namespaces": {
    "default": {
      "namespace": "default",
      "deployments": {}
    }
  },
  "Filter": "any"
}

▶ ./goldilocks summary --filter burstable | jq .
{
  "Namespaces": {
    "default": {
      "namespace": "default",
      "deployments": {}
    }
  },
  "Filter": "burstable"
}

▶ ./goldilocks summary --filter guaranteed | jq .
{
  "Namespaces": {
    "default": {
      "namespace": "default",
      "deployments": {
        "demo-basic-demo": {
          "deploymentName": "demo-basic-demo",
          "containers": {
            "basic-demo": {
              "containerName": "basic-demo",
              "lowerBound": {
                "cpu": "15m",
```
                
(output was shortened)

TL;DR - I see recommendations for guaranteed, but not any. I had thought that "any" should include the "guaranteed", but maybe I'm off here?
